objstorage/objstorageprovider/shared_obj_name.go

```diff
@@ -37,6 +40,11 @@ func sharedObjectRefName(
 	if meta.Shared.CleanupMethod != objstorage.SharedRefTracking {
 		panic("ref object used when ref tracking disabled")
 	}
+	if meta.Shared.CustomObjectName != "" {
+		return fmt.Sprintf(
+			"%s.ref.%d.%06d", meta.Shared.CustomObjectName, refCreatorID, refFileNum.FileNum(),
```

Comment:
I'm good with this implementation, but we might not need to support the combination of `CustomObjectName != "" && CleanupMethod == SharedRefTracking`.

Reply:
Agreed, it was just easy enough to include just in case.